### DISCUSSION TITLE
Side navigation: Is it possible to have multiple layers to the navigation?

### DISCUSSION THREAD

**Discussion by DanRoscigno** [markdoc/markdoc#434] [Q&A] [answered]
Note: I am a tech writer and have not written much JavaScript.

I would like to have multiple layers of navigation.  In the example below `Get started` has only one layer, but `Integrations` has two layers.  I do not see how to do this in [SideNav.js](https://github.com/markdoc/docs/blob/main/components/Shell/SideNav.js):

- Get Started
  - topic 1
  - topic 2
- Integrations
  - Visualization
    - Grafana
    - Superset
    - Preset
  - ETL
    - DBT
    - Airbyte

Thanks so much!

**Comment by mfix-stripe** [comment, marked as answer]
@DanRoscigno there are no constraints to how your side nav operates, it just requires updating the code. 

You could add an `items: []` array to each `link` object, and then add an additional `ul` here: https://github.com/markdoc/docs/blob/main/components/Shell/SideNav.js#L65

**Reply by DanRoscigno** [reply]
Thanks @mfix-stripe , I will give it a go!

**Reply by DanRoscigno** [reply]
Hi Mike, @mfix-stripe I have it working with single and double-level nav entries.  

I am calling my array of second-level items `subItems`:

I have a follow-up question if you have time:

It seems that if I have an entry in the `items` array that does not have any subitems, then I should add an empty `subLinks[]` array like so:

`{ href: '/docs/loading/Loading_intro', children: 'Understanding StarRocks table design', subLinks: []},`

I had thought about checking the length of subLinks, but if it does not exist in the current item, then I can't check for the length, so I just used an empty array.  Is that a bad idea?

Thanks again!
